packages/eslint-plugin-template/tests/rules/no-duplicate-attributes.test.ts

```diff
@@ -0,0 +1,62 @@
+import { RuleTester } from '@angular-eslint/utils';
+import rule, {
+  MessageIds,
+  RULE_NAME,
+} from '../../src/rules/no-duplicate-attributes';
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parser: '@angular-eslint/template-parser',
+});
+
+const messageId: MessageIds = 'noDuplicateAttributes';
+
+ruleTester.run(RULE_NAME, rule, {
+  valid: [
+    '<input name="foo">',
+    '<input [name]="foo">',
+    '<input (change)="bar()">',
+    '<input [(ngModel)]="foo">',
+    '<input [(ngModel)]="model" (ngModelChange)="modelChanged()">',
+  ],
+  invalid: [
+    {
+      code: '<input [name]="foo" [name]="bar">',
+      errors: [
```

Comment:
Sadly the `convertAnnotatedSourceToFailureCase` did not work here as it only supports one failing error, it might be nice in a future PR to add support for outputting multiple errors from that util.

Reply:
Ah I missed that, thank you!